clash-protocols/src/Protocols/Idle.hs

```diff
@@ -4,24 +4,31 @@
 Functionalities to easily create idle circuits for protocols.
 -}
 module Protocols.Idle (
+  -- * Type classes
   IdleCircuit (..),
+
+  -- * Utility functions
   idleSource,
   idleSink,
+  forceResetSanityGeneric,
 ) where
 
-import qualified Clash.Prelude as C
+import Clash.Prelude
+import Prelude ()
+
 import Data.Proxy
-import Protocols.Cpp (maxTupleSize)
 import Protocols.Internal
 import Protocols.Internal.TH (idleCircuitTupleInstances)
+import Protocols.Plugin
+import Protocols.Plugin.Cpp (maxTupleSize)
```

Comment:
As you can see in the generated Haddock, this does not work. `maxTupleSize` is supposed to be 3 when generating Haddock. But when generating Haddock for `clash-protocols`, `clash-protocols-base` is compiled _without_ `-DHADDOCK_ONLY` and so `maxTupleSize` is 12.

This package will need its own copy of `Cpp.hs`. That will also solve that `clash-protocols` will once again honour its own `-flarge-tuples` flag instead of the one on `clash-protocols-base`.